src/main/java/org/broadinstitute/hellbender/tools/funcotator/dataSources/gencode/GencodeFuncotationFactory.java

```diff
@@ -0,0 +1,1218 @@
+        // Get our "best" transcript:
+        final int bestTranscriptIndex = getBestTranscriptIndex(gtfFeature, variant);
+        if ( bestTranscriptIndex == -1 ) {
+            throw new GATKException("Could not get a good transcript for the given feature: " + gtfFeature.toString());
+        }
+        final GencodeGtfTranscriptFeature transcript = gtfFeature.getTranscripts().remove(bestTranscriptIndex);
+
+        final GencodeGtfFeature.GeneTranscriptType geneType = gtfFeature.getGeneType();
+
+        // We only fully process protein-coding regions.
+        // For other gene types, we do trivial processing and label them as either LINCRNA or RNA:
+        // TODO: Add more types to Variant Classification to be more explicit and a converter function to go from gene type to variant classification.
+        if ( geneType != GencodeGtfFeature.GeneTranscriptType.PROTEIN_CODING) {
+
+            // Setup the "trivial" fields of the gencodeFuncotation:
+            final GencodeFuncotation gencodeFuncotation = createGencodeFuncotationWithTrivialFieldsPopulated(variant, altAllele, gtfFeature, transcript);
+
+            if ( geneType == GencodeGtfFeature.GeneTranscriptType.LINCRNA) {
+                gencodeFuncotation.setVariantClassification(GencodeFuncotation.VariantClassification.LINCRNA);
+            }
+            else {
+                gencodeFuncotation.setVariantClassification(GencodeFuncotation.VariantClassification.RNA);
+            }
+            gencodeFuncotations.add(gencodeFuncotation);
+        }
+        else {
+            // Find the sub-feature of transcript that contains our variant:
+            final GencodeGtfFeature containingSubfeature = getContainingGtfSubfeature(variant, transcript);
+
+            // Determine what kind of region we're in and handle it in it's own way:
+            if (containingSubfeature == null) {
+
+                // We have an IGR variant
+                gencodeFuncotations.add( createIgrFuncotation(altAllele) );
+
+            } else if (GencodeGtfExonFeature.class.isAssignableFrom(containingSubfeature.getClass())) {
+
+                // We have a coding region variant
+                gencodeFuncotations.add( createCodingRegionFuncotation(variant, altAllele, gtfFeature, reference, transcript, (GencodeGtfExonFeature) containingSubfeature) );
+
+            } else if (GencodeGtfUTRFeature.class.isAssignableFrom(containingSubfeature.getClass())) {
+
+                // We have a UTR variant
+                gencodeFuncotations.add( createUtrFuncotation(variant, altAllele, reference, gtfFeature, transcript, (GencodeGtfUTRFeature) containingSubfeature) );
+
+            } else if (GencodeGtfTranscriptFeature.class.isAssignableFrom(containingSubfeature.getClass())) {
+
+                // We have an intron variant
+                gencodeFuncotations.add( createIntronFuncotation(variant, altAllele, gtfFeature, transcript) );
+
+            } else {
+
+                // Uh-oh!  Problemz.
```

Comment:
Choose a more specific GATKException.  Not sure which one, but possible ShouldNeverReachHereException

Reply:
That one sounds good.